zerver/tests/test_event_queue.py

```diff
@@ -674,6 +675,28 @@ def test_bot_recipient(self) -> None:
         self.destroy_event_queue(hambot, self.client_descriptor.event_queue.id)
         self.client_descriptor = hamlet_client_descriptor
 
+    # Internal PMs
+    def test_disable_external_notifications(self) -> None:
+        # Email and push notifications should not be sent.
+        # Currently, external notifications are disabled for PMs sent by welcome bot.
+        # In general, it should be true for any sender, with the parameter set to true.
+        msg_id = internal_send_private_message(self.iago, self.user_profile, "Test Content", True)
+        assert msg_id is not None
+        with mock.patch("zerver.tornado.event_queue.maybe_enqueue_notifications") as mock_enqueue:
+            missedmessage_hook(self.user_profile.id, self.client_descriptor, True)
+            mock_enqueue.assert_called_once()
+            args_dict = mock_enqueue.call_args_list[0][1]
+
+            self.assert_maybe_enqueue_notifications_call_args(
+                args_dict=args_dict,
+                message_id=msg_id,
+                user_id=self.user_profile.id,
+                pm_email_notify=True,
+                pm_push_notify=True,
+                disable_external_notifications=True,
+                already_notified={"email_notified": False, "push_notified": False},
```

Comment:
Does this part of the test have a different result if external notifications were not enabled? It's worth calling attention in a comment to the fact that (I think) this last `already_notified` line is False because of this parameter, since it's otherwise kinda unclear what exactly we're testing...

Reply:
> external notifications were not enabled?

You mean not disabled? Right, if they weren't disabled, the `already_notified` values would both be true. If we're going to add a comment explaining this, would it be better to add it as a docstring to this test class (since the other tests here also follow this pattern of asserting the `UserNessageNotificatioData` instance fields and also asserting the `already_notified` values)?

Reply:
Yeah, that's what I meant. Maybe we're good here even without a new comment, just wanted to make sure I was reading this right.